### PR TITLE
Updates the spess dragon antag

### PR DESCRIPTION
**Pull request by ghost**
# Document the changes in your pull request

Makes Space Dragon antag be actually up to date, since yogstation doesn't have [the rework](https://github.com/tgstation/tgstation/commit/422ef20ba8a6a754aa06d9ad0ca1913e867b14fd). This rework is fun and good, that makes the spess dragon not just being a sentient megafauna that arrived to kill everyone. Carp rifts are fun.

# Wiki Documentation

Actually yogstation wiki doesn't have a page about this antagonist, so it should be created. 

# Changelog

:cl:  
rscadd: Ports Space Dragon rework.
imageadd: Adds new fancy images for the Space Dragon and for the carp rift.  
/:cl:

### CODE REVIEW

**Comment by ghost** [comment]
Fucking hell

File: code/__DEFINES/role_preferences.dm [hunk 1]
```diff
@@ -85,9 +88,14 @@ GLOBAL_LIST_INIT(special_roles, list(
...
 	ROLE_ZOMBIE = /datum/game_mode/zombie,
-	ROLE_FUGITIVE
+	ROLE_FUGITIVE,
+	ROLE_BLOODSUCKER = /datum/game_mode/bloodsucker,
+	ROLE_MONSTERHUNTER, 
+	ROLE_SPACE_DRAGON,
 ))
 
+
+#define ROLE_SPACE_DRAGON		"Space Dragon"
```
Comment (tattax) [inline code/__DEFINES/role_preferences.dm]:
```suggestion
```

File: yogstation.dme [hunk 2]
```diff
@@ -3736,4 +3776,5 @@
 #include "yogstation\code\modules\xenoarch\loot\gigadrill.dm"
 #include "yogstation\code\modules\xenoarch\loot\guns.dm"
 #include "yogstation\interface\interface.dm"
-// END_INCLUDE
+#include "code\modules\mob\living\simple_animal\hostile\space_dragon.dm"
```
Comment (tattax) [inline yogstation.dme:3779]: add it to the correct position in the alphabetical order or atleast try to

File: code/modules/mob/living/simple_animal/hostile/space_dragon.dm [hunk 3]
```diff
@@ -0,0 +1,531 @@
...
+	/// The togglable small sprite action
+	var/small_sprite_type = /datum/action/small_sprite/megafauna/spacedragon
+	/// The innate ability to use wing gust
+	var/datum/action/innate/space_dragon/gustAttack/gust
+	/// The innate ability to summon rifts
+	var/datum/action/innate/space_dragon/summonRift/rift
+
+/mob/living/simple_animal/hostile/space_dragon/Initialize(mapload)
+	. = ..()
+	ADD_TRAIT(src, TRAIT_SPACEWALK, INNATE_TRAIT)
```
Comment (tattax) [inline code/modules/mob/living/simple_animal/hostile/space_dragon.dm]:
```suggestion
```

File: code/modules/mob/living/simple_animal/hostile/space_dragon.dm [hunk 4]
```diff
@@ -0,0 +1,531 @@
...
+	icon_living = "spacedragon"
+	icon_dead = "spacedragon_dead"
+	obj_damage = 50
+	environment_smash = ENVIRONMENT_SMASH_NONE
+	flags_1 = PREVENT_CONTENTS_EXPLOSION_1 | HEAR_1
+	melee_damage_upper = 35
+	melee_damage_lower = 35
+	armour_penetration = 30
+	pixel_x = -16
+	turns_per_move = 5
```
Comment (tattax) [inline code/modules/mob/living/simple_animal/hostile/space_dragon.dm:40]:
```suggestion
	pixel_x = -16
	turns_per_move = 5
	spacewalk = TRUE
```

File: code/__DEFINES/traits.dm [hunk 5]
```diff
@@ -165,6 +165,7 @@
 #define TRAIT_TOXINLOVER		"toxinlover"
 #define TRAIT_NOBREATH			"no_breath"
 #define TRAIT_ANTIMAGIC			"anti_magic"
+#define TRAIT_SPACEWALK         "spacewalk"
```
Comment (tattax) [inline code/__DEFINES/traits.dm]: we don't need this
```suggestion
```

File: code/__DEFINES/role_preferences.dm [hunk 6]
```diff
@@ -47,7 +47,10 @@
 #define ROLE_HORROR				"Eldritch Horror" // Yogs
 #define ROLE_INFILTRATOR		"Infiltrator" // Yogs
 #define ROLE_ZOMBIE				"Zombie"
-
+#define ROLE_BLOODSUCKER		"Bloodsucker"
+#define ROLE_VAMPIRICACCIDENT	"Vampiric Accident"
+#define ROLE_BLOODSUCKERBREAKOUT	"Bloodsucker Breakout"
+#define ROLE_MONSTERHUNTER		"Monster Hunter"
```
Comment (tattax) [inline code/__DEFINES/role_preferences.dm:53]:
```suggestion
#define ROLE_BLOODSUCKER		"Bloodsucker"
#define ROLE_VAMPIRICACCIDENT	"Vampiric Accident"
#define ROLE_BLOODSUCKERBREAKOUT	"Bloodsucker Breakout"
#define ROLE_MONSTERHUNTER		"Monster Hunter"
#define ROLE_SPACEDRAGON				"Space Dragon"
```

File: code/modules/mob/mob_movement.dm [hunk 7]
```diff
@@ -285,6 +285,8 @@
 /mob/Process_Spacemove(movement_dir = 0)
 	if(spacewalk || ..())
 		return TRUE
+	if(. || HAS_TRAIT(src, TRAIT_SPACEWALK))
+		return TRUE
```
Comment (tattax) [inline code/modules/mob/mob_movement.dm]:
```suggestion
```

**Review by nesisknesis — APPROVED** [review]

**Comment by ghost** [comment]
Coggers

**Pull request closed without merging**